tools/solution_drawer.cpp

```diff
@@ -0,0 +1,147 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <set>
+#include <vector>
+#include <random>
+
+#include <SFML/Graphics.hpp>
+#include <SFML/Window.hpp>
+
+sf::RenderWindow& initialize(double size) {
+  sf::VideoMode vm = sf::VideoMode::getDesktopMode();
+  sf::ContextSettings settings;
+  settings.antialiasingLevel = 0;
+  static sf::RenderWindow window(sf::VideoMode(vm.width / size, vm.height / size), "Vehicle Routing Problem", sf::Style::Titlebar | sf::Style::Close | sf::Style::Resize, settings);
+  window.setPosition(sf::Vector2i(vm.width / 2 - vm.width / (size * 2), vm.height / 2 - vm.height / (size * 2)));
+  window.setKeyRepeatEnabled(1);
+  window.setVerticalSyncEnabled(1);
+  return window;
+}
+
+class SolutionDrawer {
+  static constexpr auto ZOOM_FACTOR = 0.9;
+  static constexpr auto RADIUS      = 3.1;
+public:
+  SolutionDrawer(SolutionDrawer&&) = delete;
+  SolutionDrawer(SolutionDrawer const&) = delete;
+  SolutionDrawer(sf::RenderWindow& window): window {window} {}
+
+  auto solution_parse(std::string const& filename) -> void {
+
+    std::fstream fstream {filename, std::ios_base::in};
+    std::string current_line;
+    std::uint32_t route, vehicle, customer;
+    std::getline(fstream, current_line); // Skipping the solution header.
+
+    while (std::getline(fstream, current_line)) {
+      std::replace(std::begin(current_line), std::end(current_line), ';', ' '); // Replace the commas by spaces.
+      std::istringstream iss (current_line);
+      if (iss >> route >> vehicle >> customer) {
+          route_index.emplace_back(route);
+          customers_id.emplace_back(customer);
+      }
+    }
+
+    routes.resize(std::unique(std::begin(route_index), std::end(route_index)) - std::begin(route_index));
```

Comment:
I wonder if we can actually somehow highlight the same routes and not just simply ignore them. Can SFML do like double lines or a bold lines so that such routes look differently?

Reply:
I think that transparent colors may do the trick somehow as well

Reply:
by the way, can we draw customers with different demand differently? like, if customer has small demand, there's a small circle. if demand is big, the circle is big. That way we can visualize sizes as well.

there are only 5 distinct demands in Solomon's instances, I believe, so let's do it

Reply:
@andrgolubev As far as I know, there were no equal routes in the solution output.
Although, yes, we certainly can draw the customers and make the circle's size proportional to corresponding demand. These changes will be introduced soon enough, I'm going to fix it and create new pull request.